src/TextInput.tsx

```diff
@@ -1,22 +1,42 @@
 import classnames from 'classnames'
-import React from 'react'
+import React, {MouseEventHandler} from 'react'
 import {ComponentProps, Merge} from './utils/types'
 import UnstyledTextInput from './_UnstyledTextInput'
 import TextInputWrapper from './_TextInputWrapper'
+import TextInputInnerVisualSlot from './_TextInputInnerVisualSlot'
+import {useProvidedRefOrCreate} from './hooks'
 
-type NonPassthroughProps = {
+export type TextInputNonPassthroughProps = {
   className?: string
   /** @deprecated Use `leadingVisual` or `trailingVisual` prop instead */
   icon?: React.ComponentType<{className?: string}>
+  /** Whether the to show a loading indicator in the input */
+  isLoading?: boolean
+  /**
+   * Which position to render the loading indicator
+   * 'auto' (default): at the end of the input, unless a `leadingVisual` is passed. Then, it will render at the beginning
+   * 'leading': at the beginning of the input
+   * 'trailing': at the end of the input
+   **/
+  loadingIndicatorPosition?: 'auto' | 'leading' | 'trailing' // TODO: come up with a shorter name
```

Comment:
Any better recommendations for this prop name? It's so long...

Reply:
Maybe just `loaderPosition`?